production/helm/loki/templates/_helpers.tpl

```diff
@@ -240,17 +270,13 @@ Create the service endpoint including port for MinIO.
   {{- end -}}
 {{- end -}}
 
-
-{{/*
-Return if deployment mode is simple scalable
-*/}}
-{{- define "loki.deployment.isSimpleScalable" -}}
-  {{- eq .Values.loki.deploymentMode "simple-scalable" }}
+{{/* Determine if deployment is using object storage */}}
+{{- define "loki.isUsingObjectStorage" -}}
+{{- or (eq .Values.loki.storage.type "gcs") (eq .Values.loki.storage.type "s3") -}}
 {{- end -}}
 
-{{/*
-Return if deployment mode is single binary
-*/}}
-{{- define "loki.deployment.isSingleBinary" -}}
-  {{- eq .Values.loki.deploymentMode "single-binary" }}
+{{/* Configure the correct name for the memberlist service */}}
+{{- define "loki.memberlist" -}}
+{{ include "loki.name" . }}-memberlist
```

Comment:
This is a future proofing. By moving this into a function, it will make us easier to support migrations where we honor an existing memberlist DNS to enable the new deployment to join the same ring as an existing deployment. Worst case we don't need that and this remains a one line function, which I think is worth it.